compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1188,6 +1192,74 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
         }
     }
 
+  /** Returns a block assigning to `targets` the corresponding values in `rhs`.
+   *
+   *  @param targets A sequence of untyped trees on the LHS of a multiple assignment.
+   *  @param rhs The RHS of a multuple assignment.
+   */
+  private def typedPairwiseAssignments(
+      targets: List[untpd.Tree], rhs: Tree
+  )(using Context): Tree =
+    val s = mutable.ListBuffer[untpd.Tree]()
+    val u = formPairwiseAssignments(
+      s, targets, untpd.TypedSplice(rhs), rhs.tpe,
+      EmptyTermName)
+    u.map((b) => typed(b)).getOrElse(unitLiteral)
+
+  /** Appends to `statements` the assignments of `targets` to corresponding values in `rhs`.
+   *
+   *  @param statements A partially constructed sequence representing the desugaring of a
+   *  multiple assignment.
+   *  @param targets A sequence of expression representing variables to assign.
+   *  @param rhs The expression of the values to assign..
+   *  @param rhsType The type of `rhs`.
+   *  @param prefix A prefix for the name of syntactic vals created by the methods.
+   */
+  private def formPairwiseAssignments(
+      statements: mutable.ListBuffer[untpd.Tree],
+      targets: List[untpd.Tree],
+      rhs: untpd.Tree,
+      rhsType: Type,
+      prefix: TermName
+  )(using Context): Option[untpd.Block] =
+    val source = UniqueName.fresh(prefix)
+    val d = untpd.ValDef(source, untpd.TypeTree(rhsType), rhs)
+      .withSpan(rhs.span)
+      .withFlags(Synthetic)
+    statements.append(d)
+
+    rhsType.tupleElementTypes match {
+      case None =>
+        report.error(InvalidMultipleAssignmentSource(rhsType), rhs.srcPos)
+        None
+
+      case Some(e) if targets.length != e.length =>
+        report.error(MultipleAssignmentShapeMismatch(e.length, targets.length), rhs.srcPos)
+        None
```

Comment:
Rather than kill off the tree and return unit instead (or a partial result) I think you should be using `errorTree` here. So after typing you have trees typed with ErrorType